pkg/logging/rate_limited_logger.go

```diff
@@ -1,128 +1,293 @@
-	}
+	rl.logf(zapcore.InfoLevel, limiter, template, args...)
 }
 
 // Info logs the consecutive interfaces
 func (rl *RateLimitedLogger) Info(args ...interface{}) {
-	rl.Logger.Info(args...)
+	rl.logger.Info(args...)
 }
 
 // Infof logs the input template filled with argument data
 func (rl *RateLimitedLogger) Infof(template string, args ...interface{}) {
-	rl.Logger.Infof(template, args...)
+	rl.logger.Infof(template, args...)
 }
 
 // Infow logs the input message and keyValues
 func (rl *RateLimitedLogger) Infow(msg string, keysAndValues ...interface{}) {
-	rl.Logger.Infow(msg, keysAndValues...)
+	rl.logger.Infow(msg, keysAndValues...)
 }
 
 // DebugL logs a templated debug message if allowed by the rate limiter corresponding to the identifier
 func (rl *RateLimitedLogger) DebugL(limiter string, template string, args ...interface{}) {
-	if rl.allowLog(limiter) {
-		rl.Debugf(template, args...)
-	}
+	rl.logf(zapcore.DebugLevel, limiter, template, args...)
 }
 
 // Debug logs the consecutive interfaces
 func (rl *RateLimitedLogger) Debug(args ...interface{}) {
-	rl.Logger.Debug(args...)
+	rl.logger.Debug(args...)
 }
 
 // Debugf logs the input template filled with argument data
 func (rl *RateLimitedLogger) Debugf(template string, args ...interface{}) {
-	rl.Logger.Debugf(template, args...)
+	rl.logger.Debugf(template, args...)
 }
 
 // Debugw logs the input message and keyValues
 func (rl *RateLimitedLogger) Debugw(msg string, keysAndValues ...interface{}) {
-	rl.Logger.Debugw(msg, keysAndValues...)
+	rl.logger.Debugw(msg, keysAndValues...)
+}
+
+func getLogKey(limiter string, level zapcore.Level, file string, line int, payload string) string {
+	var keyWriter strings.Builder
+	keyWriter.WriteString(limiter)
+	keyWriter.WriteString("-")
+	keyWriter.WriteString(level.CapitalString())
+	keyWriter.WriteString("-")
+	keyWriter.WriteString(file)
+	keyWriter.WriteString(":")
+	keyWriter.WriteString(fmt.Sprintf("%d", line))
+	keyWriter.WriteString("-")
+	keyWriter.WriteString(payload)
+	return keyWriter.String()
+}
+
+const (
+	localFilePathPrefix = "github.com/stackrox/stackrox/"
+	filePathPrefix      = "github.com/stackrox/rox/"
+	githubPathPrefix    = "/__w/stackrox/stackrox/"
+)
+
+func getTrimmedFilePath(path string) string {
+	prefixes := []string{filePathPrefix, localFilePathPrefix, githubPathPrefix}
+	for _, prefix := range prefixes {
+		prefixToCut := strings.Index(path, prefix)
+		if prefixToCut >= 0 {
+			cutPrefixLength := prefixToCut + len(prefix)
+			return path[cutPrefixLength:]
+		}
+	}
+	return path
+}
+
+func (rl *RateLimitedLogger) registerTraceAndLog(
+	level zapcore.Level,
+	limiter string,
+	key string,
+	payload string,
+	file string,
+	line int,
+) {
+	log := newRateLimitedLog(
+		rl.logger,
+		level,
+		limiter,
+		payload,
+		file,
+		line,
+	)
+	rl.rateLimitedLogs.Add(key, log)
+	log.log()
+}
+
+func (rl *RateLimitedLogger) logf(level zapcore.Level, limiter string, template string, args ...interface{}) {
```

Comment:
why it can not be like this:

```go
func (rl *RateLimitedLogger) logf(level zapcore.Level, limiter string, template string, args ...interface{}) {
	payload := fmt.Sprintf(template, args...)
	_, file, line, _ := runtime.Caller(2)
	file = getTrimmedFilePath(file)
	key := getLogKey(limiter, level, file, line, payload)
	throttledLog, found := rl.rateLimitedLogs.Get(key)
	if !found {
		rl.registerTraceAndLog(level, limiter, key, payload, file, line)
		return
	}
	throttledLog.count.Add(1)
}
```